feat: Add faro receiver (#38224)
open-telemetry/opentelemetry-collector-contrib: receiver/faroreceiver/README.md

# Faro Receiver

This receiver can receive telemetry data from the [Grafana Faro Web SDK](https://github.com/grafana/faro-web-sdk).

> The telemetry data is in JSON format and adheres to the Faro OpenAPI schema, as defined in the [github.com/grafana/faro](https://github.com/grafana/faro)

Reviewer: ```suggestion
The telemetry data must be in JSON format and adhere to the [Faro OpenAPI schema](https://github.com/grafana/faro/blob/main/spec/gen/faro.gen.yaml).
```

## Receiver Configuration

> The following receiver configuration parameters are supported.

Reviewer: Need to reference the full available server config (https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp) not just endpoint.

Author: Agreed
